crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Upgrade
+    attr_accessor :data
+
+    def initialize(backup)
+      @backup = backup
+      @data = @backup.data
+      @version = @backup.version
+    end
+
+    def upgrade
+      knife_files
+      crowbar_files
+    end
+
+    def supported?
+      upgrades = [
+        ["1.9", "3.0"]
+      ]
+      upgrades.include?([@version, ENV["CROWBAR_VERSION"]])
+    end
+
+    protected
+
+    def knife_files
+      @data.join("knife", "data_bags").rename(@data.join("knife", "databags"))
```

Comment:
This fails for my cloud5 backup created with https://github.com/crowbar/barclamp-crowbar/pull/1359 I guess it's not needed anymore?